### PR TITLE
[BFCL] Overhaul apply_function_credential_config.py for Enhanced Usability

### PR DESCRIPTION
**Pull request by HuanzhiMao**
Currently, the `apply_function_credential_config.py` requires the user to manually provide the input file path and it only takes one single file path at a time, which is very cumbersome and user-unfriendly, considering that the user must do this 14 times for each of the test dataset files. 

This PR refactors the script to improve its usability:

1. The script no longer requires an input file path. It automatically searches for datasets in the default `./data/` directory. 
2. The script now can process all files within the specified location (when `--input-path` is set to a directory path), thereby supporting bulk operations.
3. Single file path input is still supported. 

This PR partially addresses #501 and #502.

### CODE REVIEW

File: berkeley-function-call-leaderboard/apply_function_credential_config.py [hunk 1]
```diff
@@ -61,17 +57,50 @@ def main():
...
+    for input_file_path in glob.glob(json_files_pattern):
+        file_name = os.path.basename(input_file_path)
+        output_file_path = os.path.join(output_dir, file_name)
+        process_file(input_file_path, output_file_path)
+        
+    # Process each subdirectory
+    # subdirs = [entry.path for entry in entries if entry.is_dir()]
+    # for subdir in subdirs:
+    #     process_dir(subdir, subdir)
+
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/apply_function_credential_config.py:86]: Delete commented code

**Review by CharlieJCJ — APPROVED** [review]
LGTM, 

some minor cleanup.

File: berkeley-function-call-leaderboard/apply_function_credential_config.py [hunk 2]
```diff
@@ -61,17 +57,48 @@ def main():
                 # Handle the case where a line is not a valid JSON object
                 print("Invalid JSON line skipped.")
                 continue
+            
+    # Write each modified JSON object overwrite the output file
```
Comment (ShishirPatil) [inline berkeley-function-call-leaderboard/apply_function_credential_config.py:61]: fix: grammar and writing mechanics. Unclear what this means

Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/apply_function_credential_config.py]: addressed

**Review by ShishirPatil — APPROVED** [review]

**Pull request merged**